### PR TITLE
jit: support for less than equal for floats.

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

File: wasm/jit/jit_amd64_test.go [hunk 1]
```diff
@@ -1176,6 +1176,138 @@ func TestAmd64Builder_handleLe(t *testing.T) {
...
+			x1, x2 float64
+			exp    bool
+		}{
+			{x1: 100, x2: -1.1, exp: false},
+			{x1: -1, x2: 100, exp: true},
+			{x1: 100, x2: 200, exp: true},
+			{x1: 100.01234124, x2: 100.01234124, exp: true},
+			{x1: 100.01234124, x2: -100.01234124, exp: false},
+			{x1: 200.12315, x2: 100, exp: false},
+			{x1: math.Inf(1), x2: 100, exp: false},
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64_test.go:1256]: should we add normal number that requires 64-bit precision also?

File: wasm/jit/jit_amd64.go [hunk 2]
```diff
@@ -1038,65 +1038,57 @@ func (b *amd64Builder) handleSub(o *wazeroir.OperationSub) error {
...
-		tp = generalPurposeRegisterTypeInt
+		prog.As = x86.ACMPQ
+		prog.From.Reg = x1.register
+		prog.To.Reg = x2.register
 	case wazeroir.SignedTypeFloat32:
-		tp = generalPurposeRegisterTypeFloat
-		panic("add test!")
+		resultConditionState = conditionalRegisterStateBE
+		prog.As = x86.ACOMISS
+		prog.From.Reg = x2.register
```
Comment (codefromthecrypt) [inline wasm/jit/jit_amd64.go:1080]: maybe it is intuitive to people more familiar, but I found it odd that the from/to are permuted when a float..

**Review by codefromthecrypt — APPROVED** [review]
Looking good, can't find much to suggest as it is quite clean!

**Comment by mathetake** [comment]
@codefromthecrypt thanks for the look!

**Pull request merged**